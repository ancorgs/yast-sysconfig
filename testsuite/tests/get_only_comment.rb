# encoding: utf-8

# Sysconfig module - testsuite
#
# get_only_comment tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class GetOnlyCommentClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      # Note: Testsuite::Dump is used instead of Testsuite::Test, because it supports multiline
      # output (Testsuite::Test dumps only first line)

      @input = nil
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = ""
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = "# sdfa  "
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = "### gfh "
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = "# dsfds"
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = "# adsf \n# sdfwer\n"
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))
      Testsuite.Dump("---")

      @input = "\n\n#134\n   \n#34763\n  ##Path"
      Testsuite.Dump(Builtins.sformat("Input: %1", @input))
      Testsuite.Dump("Output:")
      Testsuite.Dump(Sysconfig.get_only_comment(@input))

      nil
    end
  end
end

Yast::GetOnlyCommentClient.new.main
