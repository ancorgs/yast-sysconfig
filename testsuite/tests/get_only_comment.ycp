// Sysconfig module - testsuite
//
// get_only_comment tests
//
// testedfiles: Sysconfig.ycp
//
// $Id$
//

{

import "Testsuite";

import "Pkg";   // override packamanager

import "Mode";		// mode is imported in Product
Mode::config = true;	// package manager is initialized in Product constructor when Mode::config is false !!
import "Sysconfig";

// Note: Testsuite::Dump is used instead of Testsuite::Test, because it supports multiline
// output (Testsuite::Test dumps only first line)

string input = nil;
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "# sdfa  ";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "### gfh ";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "# dsfds";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "# adsf \n# sdfwer\n";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );
Testsuite::Dump( "---" );

input = "\n\n#134\n   \n#34763\n  ##Path";
Testsuite::Dump( sformat("Input: %1", input) );
Testsuite::Dump( "Output:" );
Testsuite::Dump( Sysconfig::get_only_comment(input) );

}

