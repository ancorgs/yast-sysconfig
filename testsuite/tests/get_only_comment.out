Dump	Input: nil
Dump	Output:
Dump	
Dump	---
Dump	Input: 
Dump	Output:
Dump	
Dump	---
Dump	Input: # sdfa  
Dump	Output:
Dump	 sdfa  
Dump	
Dump	---
Dump	Input: ### gfh 
Dump	Output:
Dump	
Dump	---
Dump	Input: # dsfds
Dump	Output:
Dump	 dsfds
Dump	
Dump	---
Dump	Input: # adsf 
Dump	# sdfwer
Dump	
Dump	Output:
Dump	 adsf 
Dump	 sdfwer
Dump	
Dump	---
Dump	Input: 
Dump	
Dump	#134
Dump	   
Dump	#34763
Dump	  ##Path
Dump	Output:
Dump	134
Dump	34763
Dump	
